src/Modules/UngradedModules.jl

```diff
@@ -3877,17 +3878,206 @@ function free_resolution(F::FreeMod)
   return res
 end
 
+#= Fill functions (and helpers) for Hecke ChainComplexes in terms of free resolutions =#
+function get_last_map_key(cc::Hecke.ChainComplex)
+  ctr = cc.start
+  while haskey(cc.maps, ctr)
+    ctr -= 1
+  end
+  return ctr+1
+end
+
+function extend_free_resolution(cc::Hecke.ChainComplex, idx::Int; algorithm::String="fres")
```

Comment:
```suggestion
function extend_free_resolution(cc::Hecke.ChainComplex, idx::Int; algorithm::Symbol=:fres)
```
... and change the rest accordingly.

Reply:
Same applies elsewhere

Reply:
Fixed.